backend/test/remove_admin_permissions_hierarchy_test.py

```diff
@@ -107,3 +107,260 @@ def test_post(self):
+
+        first_inst.put()
+        second_inst.put()
+        third_inst.put()
+        fourth_inst.put()
+
+        first_user.put()
+
+        self.assertTrue(has_permissions(
+            first_user, first_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+        self.assertTrue(has_permissions(
+            first_user, second_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+        self.assertTrue(has_permissions(
+            first_user, third_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+        self.assertTrue(has_permissions(
+            first_user, fourth_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+
+        first_user.institutions.append(first_inst.key)
+        first_user.follows.append(first_inst.key)
+        first_user.put()
+
+        params = '?institution_key=%s&&parent_key=%s' % (
+            third_inst.key.urlsafe(), second_inst.key.urlsafe())
+        self.testapp.post(
+            REMOVE_ADMIN_PERMISSIONS_URI + params
+        )
+
+        first_user = first_user.key.get()
+
+        self.assertTrue(has_permissions(
+            first_user, first_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+        self.assertTrue(has_permissions(
+            first_user, second_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+        self.assertTrue(has_permissions(
+            first_user, third_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+        self.assertTrue(has_permissions(
+            first_user, fourth_inst.key.urlsafe(), permissions.DEFAULT_ADMIN_PERMISSIONS))
+    
+    def test_with_permissions_removal_in_the_top_of_the_hierarchy(self):
+        """Test the post method."""
+        # Verify the members
+        """Test remove admin permissions in institution hierarchy."""
+        first_user = mocks.create_user()
+        second_user = mocks.create_user()
+        third_user = mocks.create_user()
+
+        first_inst = mocks.create_institution()
+        second_inst = mocks.create_institution()
+        third_inst = mocks.create_institution()
+        fourth_inst = mocks.create_institution()
+
+        first_inst.add_member(first_user)
+        first_inst.set_admin(first_user.key)
+        third_inst.add_member(second_user)
+        third_inst.set_admin(second_user.key)
+
+        second_inst.add_member(second_user)
+        second_inst.set_admin(second_user.key)
+
+        fourth_inst.add_member(third_user)
+        fourth_inst.set_admin(third_user.key)
+
+        first_user.institutions_admin.append(first_inst.key)
+        second_user.institutions_admin.append(second_inst.key)
+        second_user.institutions_admin.append(third_inst.key)
+        third_user.institutions_admin.append(fourth_inst.key)
+
+        # Hierarchy
+        #   first_inst -> second_inst -> third_inst -> fourth_inst
+        #   (first_user)  (second_user)   (second_user)  (third_user)
+        second_inst.parent_institution = first_inst.key
```

Comment:
try to change the administrator of fourth_inst to first_user and check if the permissions still ok